lib/request-handler.js

```diff
@@ -351,12 +362,25 @@ RequestHandler.prototype.handleError = function (err, callback) {
   callback(err);
 };
 
-RequestHandler.prototype.retry = function (callback) {
+/**
+ * @param {Number} consistency
+ * @param {Boolean} useCurrentHost
+ * @param {Function} callback
+ */
+RequestHandler.prototype.retry = function (consistency, useCurrentHost, callback) {
   this.retryCount++;
   this.log('info', 'Retrying request');
+  if (typeof consistency === 'number') {
+    this.request.consistency = consistency;
+  }
   if (this.retryHandler) {
+    //custom retry handler (not a QueryRequest / ExecuteRequest / BatchRequest)
     return this.retryHandler(callback);
   }
+  if (useCurrentHost !== false) {
+    // retry on the current host by default
+    return this.sendOnConnection(this.request, this.requestOptions, callback);
+  }
   this.send(this.request, this.requestOptions, callback);
```

Comment:
Ah, when I made the comment I must not have noticed the `return` in the sendOnConnection call, in that case this is fine.